src/shogun/lib/DynamicObjectArray.h

```diff
@@ -434,27 +417,14 @@ class CDynamicObjectArray : public CSGObject
 			// current_num_elements as size (see parameter
 			// registration) the cloned version has less memory
 			// allocated than known to dynarray. We fix this here.
-			cloned->m_array.num_elements = cloned->m_array.current_num_elements;
+			cloned->m_array = cloned->m_array;
 			return cloned;
 		}
 
 	private:
 		/** register parameters */
 		virtual void init()
 		{
```

Comment:
by removing all these variables the parameter framework has no way to detect whether there is an element in the `m_array` or not... this way for example:
```
CDynamicObjectArray* array1=new CDynamicObjectArray();
CDynamicObjectArray* array2=new CDynamicObjectArray();

array1->append_element(new CGaussianKernel());

EXPECT_FALSE(TParameter::compare_ptype(PT_SGOBJECT, &array1, &array2));```

will fail as the `TParameter` framework will not know anything about the size or the content of `array1` or `array2`. i reckon the best course of action in this case is actually override the `equals` function in `CDynamicObjectArray` and check the size and content of `m_array`.